src/transformers/onnx/features.py

```diff
@@ -79,7 +98,7 @@ def supported_features_mapping(
 
 
 class FeaturesManager:
-    if is_torch_available():
+    if is_torch_available() and not is_tf_available():
```

Comment:
Same here, instead of having three tests, why not always have `_TASKS_TO_AUTOMODELS` be a nested dict with frameworks, and you then fill the frameworks when each framework if available?

Reply:
That's a nice idea - thanks! In the end we may not need this if we adopt solution 2 :)